src/components/PlanetDetails/PlanetDetails.tsx

```diff
@@ -0,0 +1,95 @@
+import {
+  useLoaderData,
+  useNavigate,
+  useNavigation,
+  useParams,
+} from 'react-router-dom';
+import { Planet } from '../../services/searchService/types';
+import Spinner from '../Spinner/Spinner';
+
+import styles from './planetDetails.module.css';
+
+export default function PlanetDetails() {
+  const { planetName } = useParams();
+  const planetDetails = useLoaderData() as Planet;
```

Comment:
avoid type casting

Reply:
TODO: i need some research on how to this with useLoaderData hook.. :/